src/lib.rs

```diff
@@ -189,8 +189,7 @@
 
 #![warn(missing_docs)]
 
-/// Map implementation used by the library
-pub type Map<K, V> = rustc_hash::FxHashMap<K, V>;
+pub use internal::types::Map;
```

Comment:
Actually I think you should change the `use` in other in modules that expose the `Map` type in their type signature, and just put the `types.rs` module as a `pub` one in `lib.rs`.

Reply:
Does it mean you prefer users of the library to import
`pubgrub::types::Map` instead of `pubgrub::Map`?

Reply:
For now yes. They already import all the rest from their respective modules, so let's do it like that for the time being. Once we have more usage, we will see what things are the most often use, and we can make a `prelude` module that reexport those for a one stop shop `use pubgrub::prelude::*` or `prelude::{...}`

Reply:
Got it, makes sense to keep it consistent. Even `resolve` function is not exported from `solver` module after all, and _that_ one is supposed to be the easiest to find.